File: Solutions/2283- Check if Number Has Equal Digit Count and Digit Value/Solution.py
class Solution:
    def digitCount(self, num: str) -> bool:
        digit_count = [0] * 10

        for ch in num:
            digit_count[int(ch)] += 1

        for i in range(len(num)):
            if digit_count[i] != int(num[i]):
                return False

        return True

File: Solutions/2283- Check if Number Has Equal Digit Count and Digit Value/Solution.java
class Solution {
    public boolean digitCount(String num) {
        int[] digitCount = new int[10];

        for (char ch : num.toCharArray()) {
            digitCount[ch - '0']++;
        }

        for (int i = 0; i < num.length(); i++) {
            if (digitCount[i] != (num.charAt(i) - '0')) {
                return false;
            }
        }

        return true;
    }
}
